numpy/distutils/command/autodist.py

```diff
@@ -46,22 +46,47 @@ def check_restrict(cmd):
     return ''
 
 
-def check_compiler_gcc4(cmd):
-    """Return True if the C compiler is GCC 4.x."""
+def check_compiler_gcc(cmd):
+    """Check if the compiler is GCC."""
+
     cmd._check_compiler()
     body = textwrap.dedent("""
         int
         main()
         {
-        #if (! defined __GNUC__) || (__GNUC__ < 4)
-        #error gcc >= 4 required
+        #if (! defined __GNUC__)
+        #error gcc required
         #endif
             return 0;
         }
         """)
     return cmd.try_compile(body, None, None)
 
 
+def check_gcc_version_at_least(cmd, major, minor=0, patchlevel=0):
+    """
+    Check that the gcc version is at least the specified version."""
+
+    cmd._check_compiler()
+    version = '.'.join([str(major), str(minor), str(patchlevel)])
+    body = textwrap.dedent("""
+        int
+        main()
+        {
+        #if (! defined __GNUC__) || (__GNUC__ < %(major)d) || \\
+                (__GNUC_MINOR__ < %(minor)d) || \\
+                (__GNUC_PATCHLEVEL__ < %(patchlevel)d)
+        #error gcc >= %(version)s required
+        #endif
+            return 0;
+        }
+        """)
+    kw = {'version': version, 'major': major, 'minor': minor,
+          'patchlevel': patchlevel}
+
+    return cmd.try_compile(body % kw, None, None)
+
+
```

Comment:
`setup.py` uses `gcc -dumpversion` instead of compiling. Is there a reason to prefer one over the other? I guess both require a call to `subprocess.Popen`. `gcc -dumpversion` seems more straightforward than checking a compilation error.

Reply:
I suppose it could.  This code is mostly just cargo cult: Most of the other code in the module used compile checks, and there was an existing function here, for some long-forgotten reason, that explicitly checked for gcc >= 4.0, so this is just generalizing it.

Reply:
For these purposes one would also want `gcc -dumpfullversion`.

Reply:
I suppose one other benefit of this approach is that it's compiler-agnostic--it doesn't require any additional code to check that the compiler is in fact GCC, but will fail appropriately if it is not GCC.

Reply:
thanks for clarifying